components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/analytics/APIMgtResponseHandler.java

```diff
@@ -56,9 +59,14 @@ public APIMgtResponseHandler() {
     }
 
     public boolean mediate(MessageContext mc) {
-        
+
         if (publisher == null) {
```

Comment:
Why this is inside (publisher == null) block? What is the relationship of the publisher here?

Reply:
fixed